external/llvm-project/mlir/lib/Conversion/GPUCommon/GPUOpsLowering.cpp

```diff
@@ -181,8 +186,46 @@ GPUFuncOpLowering::matchAndRewrite(gpu::GPUFuncOp gpuFuncOp, OpAdaptor adaptor,
     }
   }
 
-  rewriter.eraseOp(gpuFuncOp);
-  return success();
+  // Get memref type from function arguments and set the noalias to
+  // pointer arguments.
+  for (const auto &en : llvm::enumerate(gpuFuncOp.getArgumentTypes())) {
+    auto memrefTy = en.value().dyn_cast<MemRefType>();
+    NamedAttrList argAttr =
+        argAttrs ? argAttrs[en.index()].cast<DictionaryAttr>() : NamedAttrList();
+
+  auto copyPointerAttribute = [&](StringRef attrName) {
+    Attribute attr = argAttr.erase(attrName);
+    if (!attr)
+      return;
+    auto remapping = signatureConversion.getInputMapping(en.index());
+    if (remapping->size > 1 &&
```

Comment:
Since that's not an obvious condition, comment that this is a proxy for the bare pointer calling convention.